hibernate-reactive-core/src/main/java/org/hibernate/reactive/mutiny/impl/MutinySessionImpl.java

```diff
@@ -175,12 +177,18 @@ public boolean contains(Object entity) {
 
 	@Override
 	public <R> Mutiny.Query<R> createNativeQuery(String sql, Class<R> resultType) {
-		return new MutinyQueryImpl<>( delegate.createReactiveNativeQuery( sql, resultType ) );
+		try {
+			delegate.getFactory().getMetamodel().entityPersister(resultType);
+			return new MutinyQueryImpl<>( delegate.createReactiveNativeQuery( sql, resultType ) );
+		}
+		catch (MappingException me) {
```

Comment:
Could you check if the persister exists rather then relying on an exception?

Reply:
If you can find me a method in Hibernate that does that. I could not.

Reply:
ok I'll amend it after this is merged.